core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -96,7 +97,10 @@ final class DefaultRequestLog implements RequestLog, RequestLogBuilder {
      * Updated by {@link #deferredFlagsUpdater}.
      */
     private volatile int deferredFlags;
+
     private final List<RequestLogFuture> pendingFutures = new ArrayList<>(4);
+
+    private final ReentrantLock reentrantLock = new ReentrantLock();
```

Comment:
Add `@GuardedBy("reentrantLock")` to `pendingFutures`?

Reply:
i added!